streams/client.go

```diff
@@ -172,10 +193,11 @@ func collectFailedEvents(res *kinesis.PutRecordsOutput, events []publisher.Event
 				continue
 			}
 			if r.ErrorCode == nil {
-				logp.NewLogger("streams").Warn("skipping failed event with unexpected state: corresponding kinesis record misses error code: ", r)
+				// logp.NewLogger("streams").Warn("skipping failed event with unexpected state: corresponding kinesis record misses error code: ", r)
 				continue
 			}
-			if *r.ErrorCode != "" {
+			if *r.ErrorCode == "ProvisionedThroughputExceededException" {
+				logp.NewLogger("streams").Debug("throughput exceeded. will retry", r)
```

Comment:
Why not just logp?

Reply:
What's the difference between this code and the switch above?

Reply:
I just kept the existing logging convention of the project, which uses a sub-logger for these messages.
This switch is per-record. The other one is for the entire api call's return code.